### PR TITLE
Document Client side aggregation

### PR DESCRIPTION
**Pull request by ogaca-dd**
Document Client side aggregation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -162,6 +162,21 @@ Windows has support for [unix domain socket](https://devblogs.microsoft.com/comm
 
 On MacOS Mojave, setting more than `2048` bytes for `StatsdMaxUnixDomainSocketPacketSize` is experimental.
 
+## Client side aggregation
+
+By default, metrics for basic types (gauges, counts, sets) are aggregated before they are sent. For example, instead of sending 3 times `my_metric:10|c|#tag1:value`, DogStatsD client sends `my_metric:30|c|#tag1:value` once. 
+For more technical details about how client-side aggregation works see #134.
```
Comment (apigirl) [inline README.md:168]: What is `#131` in this context?

File: README.md [hunk 2]
```diff
@@ -162,6 +162,21 @@ Windows has support for [unix domain socket](https://devblogs.microsoft.com/comm
...
+
+Enabling client-side aggregation has the benefit of reducing the network usage and also reducing the load for DogStatsD server (Datadog Agent).
+
+When an application sends a lot of different contexts but each context appear with a very low frequency, then enabling client-side aggregation may take more memory and more CPU. A context identifies a metric name, a tag sets and a metric type. The metric `datadog.dogstatsd.client.aggregated_context` reported by DogStatsD C# client counts the number of contexts in memory used for client-side aggregation. There is also the metric `datadog.dogstatsd.client.metrics_by_type` that represents the number of metrics submitted by the client before aggregation.
```
Comment (apigirl) [inline README.md:172]:
```suggestion
When an application sends a lot of different contexts but each context appears with a very low frequency, enabling client-side aggregation may take more memory and more CPU. A context identifies a metric name, a tag set and a metric type. The metric `datadog.dogstatsd.client.aggregated_context` reported by DogStatsD C# client counts the number of contexts in memory used for client-side aggregation. There is also the metric `datadog.dogstatsd.client.metrics_by_type` that represents the number of metrics submitted by the client before aggregation. 
```

File: README.md [hunk 3]
```diff
@@ -162,6 +162,21 @@ Windows has support for [unix domain socket](https://devblogs.microsoft.com/comm
...
+
+When an application sends a lot of different contexts but each context appear with a very low frequency, then enabling client-side aggregation may take more memory and more CPU. A context identifies a metric name, a tag sets and a metric type. The metric `datadog.dogstatsd.client.aggregated_context` reported by DogStatsD C# client counts the number of contexts in memory used for client-side aggregation. There is also the metric `datadog.dogstatsd.client.metrics_by_type` that represents the number of metrics submitted by the client before aggregation. 
+
+### Configuration
+
+The aggregation window is 2s by default and can be changed through [FlushInterval](https://github.com/DataDog/dogstatsd-csharp-client/blob/7.0.0/src/StatsdClient/ClientSideAggregationConfig.cs#L18). Note that the aggregation window on the Agent side is 10s for DogStatsD metrics. So for example, setting an aggregation window of 3s in the client produces a spike in your dashboard every 30s for counts metrics (as the third 10s bucket on the Agent is received 4 samples from the client).
```
Comment (apigirl) [inline README.md:176]:
```suggestion
The aggregation window is two seconds by default and can be changed using the [FlushInterval](https://github.com/DataDog/dogstatsd-csharp-client/blob/7.0.0/src/StatsdClient/ClientSideAggregationConfig.cs#L18). Note that the aggregation window on the Agent side is 10 seconds for DogStatsD metrics. For example, setting an aggregation window of 3s in the client produces a spike in your dashboard every 30s for counts metrics (as the third 10s bucket on the Agent is received 4 samples from the client).
```
The link needs to be changed to a number and added to the bottom of the file.

File: README.md [hunk 4]
```diff
@@ -162,6 +162,21 @@ Windows has support for [unix domain socket](https://devblogs.microsoft.com/comm
...
+
+To disable client-side aggregation set [ClientSideAggregation](https://github.com/DataDog/dogstatsd-csharp-client/blob/7.0.0/src/StatsdClient/StatsdConfig.cs#L141) to `null`.
```
Comment (apigirl) [inline README.md:178]: same link feedback

**Review by apigirl — COMMENTED** [review]
added some initial feedback and comments. This will need a full re review from someone on the docs team once we have a bit more context

File: README.md (hunk 1 above)
Comment (ogaca-dd) [reply README.md:168]: #134 is the link to the PR that introduced client side aggregation. 
Note: Github will create a full link from `#134`

File: README.md (hunk 3 above)
Comment (ogaca-dd) [reply README.md]: Out of curiosity, what are the pros of using a number versus what I did?

Comment (ogaca-dd) [reply README.md]: @apigirl : Should I change all the links in the REAMDE.md or is it OK to keep `text](link)` syntax?

**Review by apigirl — APPROVED** [review]

**Pull request merged**